Assets/Scripts/UI/LoadingUI.cs

```diff
@@ -31,12 +31,46 @@ public class LoadingUI : MonoBehaviour {
         if (scenesLoading.Count != 0) StartCoroutine(GetSceneLoadProgress());
     }
 
+    public void GotoScene(string sceneName) {
+        //SaveSceneToPref(sceneName);
+        this.gameObject.SetActive(true);
+        AddSceneToLoad(sceneName);
+        StartLoad();
+    }
+
+    public void SaveSceneToPref(string sceneName) {
+        int currSceneIndex = 0;
+        int latestSceneIndex = PlayerPrefs.HasKey("latestSceneIndex") ? PlayerPrefs.GetInt("latestSceneIndex") : 1;
+
+        if (sceneName == MainMenu.staticMapInfos[0].dialogueSceneName) {
+            currSceneIndex = 1;
+        } else if (sceneName == MainMenu.staticMapInfos[0].gameSceneName) {
+            currSceneIndex = 2;
+        } else if (sceneName == MainMenu.staticMapInfos[1].dialogueSceneName) {
+            currSceneIndex = 3;
+        } else if (sceneName == MainMenu.staticMapInfos[1].gameSceneName) {
+            currSceneIndex = 4;
+        } else if (sceneName == MainMenu.staticMapInfos[2].dialogueSceneName) {
+            currSceneIndex = 5;
+        } else if (sceneName == MainMenu.staticMapInfos[2].gameSceneName) {
+            currSceneIndex = 6;
+        } else if (sceneName == MainMenu.staticMapInfos[3].dialogueSceneName) {
+            currSceneIndex = 7;
+        } else if (sceneName == MainMenu.staticMapInfos[3].gameSceneName) {
+            currSceneIndex = 8;
+        } else if (sceneName == "story-end") {
+            currSceneIndex = 9;
+        }
```

Comment:
shouldn't be so messy
```suggestion

        for (int i = 0; i < MainMenu.staticMapInfos.Length; i++) {
             if (sceneName == MainMenu.staticMapInfos[i].dialogueSceneName) {
                 currSceneIndex = GetDialogueSceneIndex(i);
             } else if (if (sceneName == MainMenu.staticMapInfos[i].gameSceneName) {
                 currSceneIndex = GetGameSceneIndex(i);
             }
        }
        
```

later on include 
```c#
int GetDialogueSceneIndex(int lvl) => lvl *2 + 1;
int GetGameSceneIndex(int lvl) => lvl *2 + 1;
```

Reply:
https://github.com/CS3247-Game-Development-Team-6/Doodles/pull/344/commits/247cad910b49a99fb03667daf52c083426aeaef6